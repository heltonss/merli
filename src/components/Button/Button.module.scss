@import '../../assets/scss/colors';
@import '../../assets/scss/mixins';

.container {
  @include fontSize(16px);
  background: $blue;
  cursor: pointer;
  color: #fff;
  border: none;
  border-radius: 4px/4px;
  height: 40px;
  width: 100%;
  max-width: 225px;
}
