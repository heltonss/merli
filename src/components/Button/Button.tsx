import classNames from 'classnames';
import React from 'react';
import { Link } from 'react-router-dom';
import Style from './Button.module.scss';

interface IOwnProps {
  label: string;
  route?: string;
  func?: any;
  parentClassStyle?: string;
}

const Button: React.FC<IOwnProps> = ({ label, route, func, parentClassStyle = '' }: IOwnProps) => {
  return route ? (
    <Link className={classNames([Style.container, parentClassStyle])} to={route}>
      {label}
    </Link>
  ) : (
    <button onClick={func} className={Style.container}>
      {label}
    </button>
  );
};

export default Button;
