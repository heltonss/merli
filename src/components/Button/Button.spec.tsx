import { shallow } from 'enzyme';
import React from 'react';
import Button from './Button';

describe('Button', () => {
  it('should render without crash', () => {
    const func = () => {
      return;
    };

    const wrapper = shallow(<Button label={'label do button'} func={func} />);
    expect(wrapper).toMatchSnapshot();
  });
});
