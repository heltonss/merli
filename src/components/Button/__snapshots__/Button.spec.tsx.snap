// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render without crash 1`] = `ShallowWrapper {}`;
